### PR TITLE
Create default network as config-only when services have host networking

### PR DESCRIPTION
**Pull request by cywang117**
This eliminates chances of host-Docker address collision for apps such as the Supervisor where all services have host networking.

Context: [internal thread](https://balena.zulipchat.com/#narrow/stream/345889-loop.2Fbalena-os/topic/Supervisor.20v13.2B.20default.20Docker.20network)

Closes: #2062
Change-type: patch
Signed-off-by: Christina Ying Wang <christina@balena.io>

### CODE REVIEW

File: test/unit/compose/app.spec.ts [hunk 1]
```diff
@@ -602,6 +602,65 @@ describe('compose/app', () => {
...
+			const steps = current.nextStepsForAppUpdate(defaultContext, target);
+
+			const [createNetworkStep] = expectSteps('createNetwork', steps);
+			expect(createNetworkStep)
+				.to.have.property('target')
+				.that.has.property('config')
+				.that.deep.includes({ configOnly: true });
+		});
+
+		it('should not create a config-only network if network_mode: host is not specified for any service', async () => {
```
Comment (pipex) [inline test/unit/compose/app.spec.ts:636]: Maybe you could add a third test here for the "empty release" case, to ensure that a default network is created with `configOnly: true`?

Comment (cywang117) [reply test/unit/compose/app.spec.ts:636]: Good shout! Will do

**Comment by cywang117** [comment]
@balena-ci I self-certify!
